packages/ckeditor5-undo/src/basecommand.js

```diff
@@ -42,7 +42,26 @@ export default class BaseCommand extends Command {
 		// Refresh state, so the command is inactive right after initialization.
 		this.refresh();
 
-		this.listenTo( editor.data, 'set', () => this.clearStack() );
+		// Set the transparent batch for the `editor.data.set()` call if the
+		// batch type is not set and the `options.clearUndoStack` is falsy.
+		this.listenTo( editor.data, 'set', ( evt, data ) => {
+			const options = data[ 1 ] || {};
+
+			if ( options.batchType ) {
+				return;
+			}
+
+			options.batchType = options.supportUndo ? 'default' : 'transparent';
```

Comment:
TBH, the `supportUndo` option does not sound good in the `Undo/Redo` context. I'd rename the property for something like `disableClearingStacks` or something that will explain what it does .

Reply:
The problem with the `disableClearingStacks` is that we need to do two things here. First, we need to preserve stacks, and second, we need to set the batch type.

But this option makes sense from the end-user point of view when someone calls `edtior.set.data( { supportUndo: true } )`. Or `editor.setData( { supportUndo: true } )` 

You don't want to write `editor.data.set( { disableClearingStacks: true } )` as you would have no idea what it is about.

Reply:
The problem with the `disableClearingStacks` is that we need to do two things here. First, we need to preserve stacks, and second, we need to set the batch type.

But this option makes sense from the end-user point of view when someone calls `edtior.set.data( { supportUndo: true } )`. Or `editor.setData( { supportUndo: true } )` 

You don't want to write `editor.data.set( { disableClearingStacks: true } )` as you would have no idea what it is about.

Reply:
I understood, but still, it does not sound good: `editor.setData( data, { supportUndo: true } )` looks like the Undo feature is not supported if the option is not specified. 

What about the option called `createUndoStep: true`, which is false by default? This explicitly says what we want to achieve – "set data and create an undo step for this action".

Reply:
IMO we could either have `supportUndo` or both - `createUndoStep` + `preserveUndoHistory`. But TBH, setting only one from this pair doesn't make sense to me :smile: .

Reply:
What about:

1. `editor.data.set( { withUndo: true } )`: generic flag name to toggle an `Undo` feature.
2. `editor.data.set( { allowUndo: true } )`: as above.
3. `editor.data.set( { addUndoStep: true } )`: As @ma2ciek explained, the `Undo` feature consists of two parts (creating an undo step and preserving undo history), but IMHO from end-user perspective it would be sufficient to mention here only the "step" part in the flag name. Moreover, the "undo step" phrase is quite commonly used in our docs so far in the context of "undoing" an operation.

I vote for 2) or 3) 😄

Reply:
(2) sounds the best from all proposals to me.

Reply:
132f9f9#diff-cfbce7194f50d3984cd54ed91d7d0fc7f47dd801cc80e6487154e3ae0c0fd80cR393 :D :joy:  

For me, all these names lose some details, but I can't find a better one :)

Reply:
what about `batchType` that we talked about? it could default to unset (‘default’) and get overridden by undo plugin to `transparent` if not set to any other value.